lib/web_apis/web_apis.es6.js

```diff
@@ -0,0 +1,253 @@
+        given version of browser. The function will do nothing if the given
+        browser already exists in browserAPIs`,
+      args: [
+        {
+          name: 'browserName',
+          documentation: `The name of imported browser.`,
+          typeName: 'String',
+        },
+        {
+          name: 'browserVersion',
+          documentation: `The version of imported browser.`,
+          typeName: 'String',
+        },
+        {
+          name: 'osName',
+          documentation: `The name of operating system.`,
+          typeName: 'String',
+        },
+        {
+          name: 'osVersion',
+          documentation: `The version of operating system.`,
+          typeName: 'String',
+        },
+        {
+          name: 'apiCatalog',
+          documentation: `The web catalog of this version of
+            browser, tested on the given operating system and version.
+            The json object should be the JSON object returned from
+            com.web.catalog.apiExtractor`,
+          typeName: 'JSON',
+        },
+      ],
+      code: function(browserName, browserVersion,
+        osName, osVersion, apiCatalog) {
+        let browserKey = `${browserName}_${browserVersion}_${osName}_${osVersion}`;
+        return this.browserAPIs.select(this.GroupBy.create({
+          arg1: this.BrowserAPI.BROWSER_KEY,
+          arg2: this.Count.create(),
+        })).then((browserKeyGroups) => {
+          // Exits early if given browser already exists in browserAPIs.
+          if (browserKeyGroups.groupKeys.indexOf(browserKey) >= 0) {
+            return false;
+          }
+          let interfaceNames = Object.keys(apiCatalog);
+          for (let i = 0; i < interfaceNames.length; i++) {
+            let interfaceName = interfaceNames[i];
+            for (let j = 0; j < apiCatalog[interfaceName].length; j++) {
+              let apiName = apiCatalog[interfaceName][j];
+              this.browserAPIs.put(this.BrowserAPI.create({
+                browserName,
+                browserVersion,
+                osName,
+                osVersion,
+                interfaceName,
+                apiName,
+              }));
+            }
+          }
+          return true;
+        });
+      },
+    },
+    {
+      name: 'toMap_',
+      documentation: `A method that takes an array of browserAPIs and an
+        array of browser keys, produce a JSON containing browser and
+        interface relationship.`,
+      args: [
+        {
+          name: 'browserAPIs',
+          typeName: 'BrowserAPI Array',
+          documentation: `An array returned from DAO.select.`,
+        },
+        {
+          name: 'browserKeys',
+          typeName: 'StringArray',
+          documentation: `A list of browser keys that contained in the
+            browserAPIs.`,
+        },
+      ],
+      returns: {
+        typeName: 'JSON',
+        documentation: `Returns a JSON of the form:
+            {interfaceName: {apiName: [Boolean ...]} ...}
```

Comment:
changed the output to {interfaceName: {apiName: {BrowserKey: true, ...} ...} ... } ...}